resources/c++/typedefs.h

```diff
@@ -0,0 +1,83 @@
+#ifndef TYPEDEF_HEADER_87945
+#define TYPEDEF_HEADER_87945
+
+enum Food { Kartoffel, Sauerkraut, Bratwurst, Brezel, Bauernbrot };
+
+// Scoped enum:
+enum struct Scoped_Enum { Pizza, Pasta, Schnitzel, Backfisch };
+// Opaque scoped enum:
+enum struct Scoped_Enum_2: int;
+
+// Should not count as enum, as it defines just an alias.
+typedef enum Food Food;
+// The following does not work:
```

Comment:
What do you mean by that exactly?

Reply:
It does not compile. I will make that more specific.